packages/core/src/browser/preferences/preference-provider.ts

```diff
@@ -33,12 +33,9 @@ export class PreferenceProvider implements Disposable {
      */
     protected readonly _ready = new Deferred<void>();
 
-    constructor() {
-        this.toDispose.push(this.onDidPreferencesChangedEmitter);
```

Comment:
it is a proper pattern to collect everything to dispose in one collection and dispose only this collection

Reply:
Yes, totally agree with you. 

The reason behind this change is that I don't want `PreferenceProvider.onDidPreferencesChangedEmitter` to be disposed when `WorkspaceService.onSavedLocationChanged` happens. Please see [here](https://github.com/theia-ide/theia/pull/2941/files#diff-7fc403782f8b8d22c7dc3fe1aaade33cR32) -
I am trying to dispose everything except for that emitter. 

Could you please advise a better way to achieve my goal ? Thank you !